code/network/multiui.cpp

```diff
@@ -234,13 +234,11 @@ void multi_common_split_text()
 	int	n_chars[MAX_BRIEF_LINES];
 	const char	*p_str[MAX_BRIEF_LINES];
 
-	n_lines = split_str(Multi_common_all_text, Multi_common_text_coords[gr_screen.res][2], n_chars, p_str, MULTI_COMMON_TEXT_MAX_LINES, MULTI_COMMON_TEXT_META_CHAR);
+	n_lines = split_str(Multi_common_all_text, Multi_common_text_coords[gr_screen.res][2], n_chars, p_str, MULTI_COMMON_TEXT_MAX_LINES, MULTI_COMMON_TEXT_MAX_LINE_LENGTH, MULTI_COMMON_TEXT_META_CHAR);
 	Assert(n_lines != -1);
 
 	for ( i = 0; i < n_lines; i++ ) {
-		//The E -- This check is unnecessary, and will break when fonts that aren't bank gothic are used
```

Comment:
Just as a note as to why I removed this comment and readded the Assertion:
The assert here does _not_ check for string width in pixels or anything. 
It just checks that the string, in bytes, fits the buffer. Which is an essential Assertion, and that this is missing in other places where this string split was used is the reason that it didn't assert but CTD.
While the new function should always avoid this case, it's still a sensible assert, should that function ever be reworked

Reply:
Good reasoning, but you should test this PR on non-BankGothic fonts to make sure you're not missing anything.

Reply:
Tbh, testing this very line is hard for me, as it's a part of multi and I have no clue how to actually get into a mission with a specific mod to test this assertion.
However, I would advocate for that being unnecessary anyways:
definition of the array we srncpy into:
``char Multi_common_text[MULTI_COMMON_TEXT_MAX_LINES][MULTI_COMMON_TEXT_MAX_LINE_LENGTH];``
definition of the assert
``Assert(n_chars[i] < MULTI_COMMON_TEXT_MAX_LINE_LENGTH); ``
strncpy in question:
``strncpy(Multi_common_text[i], p_str[i], n_chars[i]);``

Even if, somehow, that Assert had anything to do with linesize, if it would be triggered here (and be commented out), it would _always_ cause a CTD, unless that string to be copied contains magical \0's. And an assert (with broken fonts) is always better than a CTD (with broken fonts).

Reply:
Ok